addons/block_code/ui/blocks/block/block.gd

```diff
@@ -61,6 +64,17 @@ func get_instruction_node() -> InstructionTree.TreeNode:
 	return node
 
 
+func update_resources(undo_redo: EditorUndoRedoManager):
+	if resource == null:
+		resource = SerializedBlockTreeNode.new()
+		resource.serialized_block = SerializedBlock.new(get_block_class(), get_serialized_props())
```

Comment:
I think you can return early here. There's no need to compare the serialized properties again if the serialized block was just constructed with them.